app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadAdapter.kt

```diff
@@ -170,11 +170,14 @@ class ThreadAdapter(
         val item = items[position]
         if (item is MessageUi && holder is MessageViewHolder) with(holder.binding) {
             when (payload) {
-                NotifyType.TOGGLE_LIGHT_MODE -> holder.handleToggleLightModePayload(item.message.uid)
-                NotifyType.RE_RENDER -> reloadVisibleWebView()
-                NotifyType.FAILED_MESSAGE -> handleFailedMessagePayload(item.message.uid)
-                NotifyType.ONLY_REBIND_CALENDAR_ATTENDANCE -> handleCalendarAttendancePayload(item.message)
-                NotifyType.ONLY_REBIND_EMOJI_REACTIONS -> handleEmojiReactionPayload(item)
+                NotifyType.ToggleLightMode -> holder.handleToggleLightModePayload(item.message.uid)
+                NotifyType.ReRender -> reloadVisibleWebView()
+                NotifyType.FailedMessage -> handleFailedMessagePayload(item.message.uid)
+                NotifyType.OnlyRebindCalendarAttendance -> handleCalendarAttendancePayload(item.message)
+                NotifyType.OnlyRebindEmojiReactions -> handleEmojiReactionPayload(item)
+                is NotifyType.MessagesCollapseStateChanged -> {
+                    holder.handleItemsCountChangedPayload(item.message, isCollapsable = payload.isCollapsable)
```

Comment:
I'd rename the `handleItemsCountChangedPayload` method name for something more specific